overlord/snapstate/autorefresh.go

```diff
@@ -172,20 +173,35 @@ func (m *autoRefresh) refreshScheduleWithDefaultsFallback() (ts []*timeutil.Sche
 		return nil, "managed", nil
 	}
 
-	refreshScheduleStr := defaultRefreshSchedule
 	tr := config.NewTransaction(m.state)
-	err = tr.Get("core", "refresh.schedule", &refreshScheduleStr)
+	err = tr.Get("core", "refresh.timer", &scheduleAsStr)
 	if err != nil && !config.IsNoOption(err) {
 		return nil, "", err
 	}
 
-	refreshSchedule, err := timeutil.ParseLegacySchedule(refreshScheduleStr)
+	if scheduleAsStr == "" {
+		logger.Noticef("refresh.timer is not set, fallback to legacy refresh.schedule")
+		// try legacy refresh schedule config option
+		err = tr.Get("core", "refresh.schedule", &scheduleAsStr)
+		if err != nil && !config.IsNoOption(err) {
```

Comment:
We should probably also check for `config.IsNoOption(err)` here and not try to `timeutil.ParseLegacySchedule()` in this case because then the string is empty and we get an error which will trigger another log message.

Reply:
Yeah, there's an issue here, but we cannot just return either. If there is no option, we need to make sure we're then using the default from the modern field instead.

Also, if this logic is indeed bogus, it means we're missing a test that verifies one of the important use cases, which is the default trickery.